src/rust/inetstack/protocols/tcp/established/sender.rs

```diff
@@ -314,31 +317,36 @@ impl Sender {
         }
     }
 
-    pub fn pop_one_unsent_byte(&self) -> Option<Buffer> {
+    pub fn pop_one_unsent_byte(&self) -> Option<DemiBuffer> {
         let mut queue = self.unsent_queue.borrow_mut();
 
         let buf = queue.front_mut()?;
         let mut cloned_buf = buf.clone();
         let buf_len: usize = buf.len();
 
         // Pop one byte off the buf still in the queue and all but one of the bytes on our clone.
-        buf.adjust(1);
-        cloned_buf.trim(buf_len - 1);
+        buf.adjust(1).expect("'buf' should contain at least one byte");
+        cloned_buf
+            .trim(buf_len - 1)
+            .expect("'cloned_buf' should contain at least one less than its professed length");
 
         Some(cloned_buf)
     }
 
-    pub fn pop_unsent(&self, max_bytes: usize) -> Option<Buffer> {
+    pub fn pop_unsent(&self, max_bytes: usize) -> Option<DemiBuffer> {
         // TODO: Use a scatter/gather array to coalesce multiple buffers into a single segment.
         let mut unsent_queue = self.unsent_queue.borrow_mut();
-        let mut buf: Buffer = unsent_queue.pop_front()?;
+        let mut buf: DemiBuffer = unsent_queue.pop_front()?;
         let buf_len: usize = buf.len();
 
         if buf_len > max_bytes {
-            let mut cloned_buf = buf.clone();
+            let mut cloned_buf: DemiBuffer = buf.clone();
 
-            buf.adjust(max_bytes);
-            cloned_buf.trim(buf_len - max_bytes);
+            buf.adjust(max_bytes)
+                .expect("'buf' should contain at least 'max_bytes'");
```

Comment:
You mean to convert the actual value to text?  I don't think you can do that trivially with `expect()`, as it just takes a `&str` (a borrowed string slice).  To have it print out the value of max_bytes, you'd have to have code outside of here prepare a String or other buffer, fill it with the message you wanted to print on error, and then pass that buffer as a slice to `expect()`.  Which would be a lot of extra code to execute in the non-error case just to pretty up this error, and also make the code harder to read, I think.

If `expect()` took a format string, then yes, I agree it would be better to print the value of `max_bytes`, rather than the name.